### DISCUSSION TITLE
How long do I need to keep sudo permissions?

### DISCUSSION THREAD

**Discussion by CTMacUser** [Homebrew/discussions#1500] [Getting started]
Been using Homebrew for many years on my Intel-based Macs.  Just got an M1 MacBook Air a couple of weeks ago and installed Homebrew just now.  I didn't import any old accounts nor files.

I read on many occasions about not making your default account an administrative one.  I did just that on my new MBA; making the first account a throw-away one, which then I create a standard account with my Apple-ID, _etc_.  Although Homebrew mentions not using `sudo`, that doesn't seem to apply to the first install, where it insists on that calling user having permissions to `sudo`.  I hacked my local ID into the "sudoers" file before installing Homebrew.  Does my account still need `sudo` access for future calls to Homebrew?  Or can I remove it from "sudoers"?

(Also, why was the default installation directory changed?)

**Comment by SMillerDev** [comment]
> Does my account still need sudo access for future calls to Homebrew? Or can I remove it from "sudoers"?

Any usage of casks or services _could_ need sudo.

> (Also, why was the default installation directory changed?)

Because otherwise we couldn't install both native and Rosetta formula.

**Reply by CTMacUser** [reply]
It probably wouldn't happen because the new architecture is first-party, but using two completely different sub-trees of the UNIX directories to store the two current architectures doesn't seem scalable in case a third (or more) happens.

**Reply by SMillerDev** [reply]
> doesn't seem scalable in case a third (or more) happens.

It isn't, but having 2 architectures in one system is already exceptional. I don't think we have to be worried about Apple making Rosetta 3 that allows ARM to ppc to x86 or something.

**Comment by sroby** [comment]
Using a separate account for admin rights on a single-user Mac is vastly overkill, imho. You're only making things more complicated for very little security benefit.
